api/jobs/handlers.py

```diff
@@ -425,6 +427,35 @@ def get_config(self, _id):
 
         if c.get('config') is not None and c.get('inputs') is not None:
             # New behavior
+
+            # API keys are only returned in-flight, when the job is running, and not persisted to the job object.
+            if j.state == 'running':
+                gear = get_gear(j.gear_id)
+
+                for key in gear['gear']['inputs']:
+                    input = gear['gear']['inputs'][key]
+
+                    if input['base'] == 'api-key':
+                        if j.origin['type'] != 'user':
+                            raise Exception('Cannot provide an API key to a job not launched by a user')
+
+                        uid = j.origin['id']
```

Comment:
A lot of times a job isn't triggered by a user, we'll want to make sure the job was triggered by a real user (`origin['type'] = 'job`) before assigning it an API key. We can re-evaluate if devices should be able to use the SDK via an API in the future but currently it is expected to be a user.

Reply:
I agree with what you're saying. Maybe you missed the user check above this line?

Reply:
I did.